lib/middleware.js

```diff
@@ -243,7 +243,7 @@ function checkImports(path, fn) {
 
 function compare(pathA, pathB) {
   pathA = pathA.split(sep);
-  pathB = pathB.split(sep);
+  pathB = pathB.split('/');
```

Comment:
pathB is always a url path.  This will never be OS dependent if I understand correctly.  Therefore we should only split on the url path separator `/`